src/main/java/com/oltpbenchmark/benchmarks/templated/TemplatedBenchmark.java

```diff
@@ -0,0 +1,287 @@
+    protected void initClassLoader() {
+        super.initClassLoader();
+
+        this.classLoader = ClassLoader.getSystemClassLoader();
+        if (workConf.getXmlConfig().containsKey("query_templates_file")) {
+            this.classLoader = this.loadQueryTemplates(
+                    workConf.getXmlConfig().getString("query_templates_file"));
+        }
+    }
+
+    @Override
+    protected Package getProcedurePackageImpl() {
+        return (GenericQuery.class.getPackage());
+    }
+
+    @Override
+    protected List<Worker<? extends BenchmarkModule>> makeWorkersImpl() {
+        List<Worker<? extends BenchmarkModule>> workers = new ArrayList<>();
+
+        try {
+            final Map<Class<? extends Procedure>, TraceTransactionGenerator> generators = new HashMap<>();
+            // Create potential parameter bindings for each template. Add those
+            // to a trace transaction generator that will determine how the
+            // parameters are used.
+            for (Entry<TransactionType, Procedure> kv : getProcedures().entrySet()) {
+                // Sanity check that the procedure has the right type.
+                if (!(kv.getValue() instanceof GenericQuery)) {
+                    continue;
+                }
+                GenericQuery proc = (GenericQuery) kv.getValue();
+                QueryTemplateInfo info = proc.getQueryTemplateInfo();
+
+                // Parse parameter values and add each combination to a generator.
+                List<GenericQueryOperation> list = new ArrayList<>();
+                int[] paramsTypes = info.getParamsTypes();
+                for (String binding : info.getParamsValues()) {
+                    CSVParser parser = new CSVParserBuilder()
+                            .withQuoteChar('\'')
+                            .build();
+                    Object[] params = parser.parseLine(binding);
+                    assert paramsTypes.length == params.length;
+                    list.add(new GenericQueryOperation(params));
+                }
+                generators.put(proc.getClass(), new TraceTransactionGenerator(list));
+            }
+
+            // Create workers.
+            int numTerminals = workConf.getTerminals();
+            LOG.info(String.format("Creating %d workers for templated benchmark", numTerminals));
+            for (int i = 0; i < numTerminals; i++) {
+                workers.add(new TemplatedWorker(this, i, generators));
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException("Unable to create workers", e);
+        }
+        return workers;
+    }
+
+    @Override
+    protected Loader<TemplatedBenchmark> makeLoaderImpl() {
+        // Loader not currently supported.
+        throw new UnsupportedOperationException();
+    }
+
+    private ClassLoader loadQueryTemplates(String file) {
+        // Instantiate Java compiler.
+        CustomClassLoader ccloader = new CustomClassLoader(this.classLoader);
```

Comment:
Thought: If you included an XML schema file for the templates could you make use of that to auto load a class from the XML?
If nothing else, it might be good for data validation prior to parsing the content into an executable class.

Reply:
Not sure I understand this comment, could you elaborate?

Reply:
I think this is an example:
https://github.com/cmu-db/benchbase/blob/main/src/main/java/com/oltpbenchmark/api/StatementDialects.java#L128
https://github.com/cmu-db/benchbase/blob/main/src/main/resources/dialect.xsd

Reply:
Added, but that was no fun!